js/background.js

```diff
@@ -200,10 +204,12 @@ function updateBadge(tabId, numBlocked){
 }
 
 chrome.tabs.onUpdated.addListener(function(id, info, tab) {
+    // an existing tab has been reloaded or changed url, clear tab data and set status
     if(tabs[id] && info.status === "loading" && tabs[id].status !== "loading"){
         tabs[id] = {'trackers': {}, "total": 0, "dispTotal": 0, 'url': tab.url, "status": "loading"};
         updateBadge(id, 0);
     }
+    // existing tab, update the status and url
     else if(tabs[id] && info.status === "complete"){
         tabs[id].status = "complete";
```

Comment:
The badge is set to zero in the `if` above and below when the tab is created. After that it's updated when anything new is blocked in the onBeforeRequest listener.